src/Alias.ts

```diff
@@ -93,16 +97,18 @@ function applyAliases(origCommand: string, depth = 0, currentlyProcessingAliases
   // First get non-global aliases, and recursively apply them
   // (unless there are any reference loops or the reference chain is too deep)
   const localAlias = Aliases.get(commandArray[0]);
-  if (localAlias && !currentlyProcessingAliases.includes(localAlias)) {
-    const appliedAlias = applyAliases(localAlias, depth + 1, [commandArray[0], ...currentlyProcessingAliases]);
+  const localrule = commandArray[0] + "->" + localAlias + "(l)";
+  if (localAlias && !currentlyProcessingAliases.includes(localrule)) {
+    const appliedAlias = applyAliases(localAlias, depth + 1, [localrule, ...currentlyProcessingAliases]);
```

Comment:
Optional: Using an array for this is kinda cruddy. It would be pretty simple/faster to switch to an object (everything here has string keys) with set = `currentlyProccesingAliases[localrule] = true` and test being `&& !currentlyProcessingAliases[localrule]`

Reply:
I'l research that. Lists are normally faster than Hashmaps / Sets for small n though, but you are definitly right for larger n. I'm interested in checking how js objects internally work, but they probably are just Hashmaps. Syntax-wise I prefer `!currentlyProcessingAliases.includes(localrule)` but from the semantics I'd prefer the hashing to stay in constant time. So I'll come back to this, when I know more.

Reply:
If you like that syntax better you could also use a Set, which tend to be slightly slower than objects (for this purpose) on some browsers but only slightly (these days).

Reply:
Just a few comments:
Sets aren't always faster than Maps, or faster than objects in this purpose. It heavily depends on browser, also version. Ultimately, only benchmarking will tell you.
Also, Objects definitely are *not* just implemented as Maps under the covers; there's a lot more going on there. JS interpreter/compilers have some crazy stuff going on under the hood.

None of that changes this though; I'm fine with using Set here.